lib/yard/handlers/ruby/decorator_handler_methods.rb

```diff
@@ -0,0 +1,132 @@
+# Helper methods to assist with processing decorators.
+module YARD::Handlers::Ruby::DecoratorHandlerMethods
+
+  # To be used in a handler's process block.
+  # Takes care of parsing method definitions passed to decorators
+  # as parameters, as well as parsing chained decorators.
+  #
+  # @yieldparam method [YARD::CodeObjects::MethodObject] Method being decorated.
+  # @yieldparam node [YARD::Parser::Ruby::AstNode] AST node of the decorated method.
+  # @yieldparam name [Symbol] Name of the decorated method.
+  #
+  # @return [Array<Hash>] Array of hashes containing :method, :node, :name.
+  #   See yield params.
+  #
+  # @param nodes [YARD::Parser::Ruby::AstNode] AST nodes that refer to decorated methods, like
+  #   indexes of statement.parameter. Defaults to all parameters. Pass nil to specify zero
+  #   parameters.
+  #
+  # @option opts [:instance, :class] :scope Scope to use for each MethodObject.
+  #   Defaults to :instance.
+  #
+  # @option opts [true, false] :transfer_docstring Set false to disable transferring the
+  #   decorator docstring to method definitions passed to the decorator as parameters.
+  #
+  # @option opts [true, false] :transfer_source Set false to disable transferring the
+  #   decorator source code string to method definitions passed to the decorator as parameters.
+  #
+  # @example
+  #   # Simply pass the method docs through to the method definition.
+  #   process do
+  #     process_decorator
+  #   end
+  #
+  # @example
+  #   # Set the decorated method's visibility to private.
+  #   process do
+  #     process_decorator :scope => :class do |method|
+  #       method.visibility = :private if method.respond_to? :visibility
+  #     end
+  #   end
+  #
+  # @!method process_decorator(*nodes, opts={}, &block)
+  def process_decorator(*nodes, &block)
+    opts = nodes.last.is_a?(Hash) ? nodes.pop : {}
```

Comment:
Would it make more sense here to use `.responds_to?(:[])` instead of `.is_a?(Hash)`? What if the implementation changes to use a Hash-like object not descended from Hash?